### DISCUSSION TITLE
Request for Assistance with Adding User Mailbox Request to graph-codegen in Rust

### DISCUSSION THREAD

**Discussion by M-Patrone** [sreeise/graph-rs-sdk#393] [General]
Hello,

I am very interested in utilizing your library to retrieve the [user mailbox](https://learn.microsoft.com/en-us/graph/api/user-get-mailboxsettings?view=graph-rest-1.0&tabs=http) from the Microsoft Graph API. As a beginner in Rust programming, I would like to understand the steps required to add this request to "graph-codegen" so that it can generate the necessary methods. Alternatively, please let me know what steps I need to take manually to add a new API resource.

Thank you for your time and assistance! I am looking forward to working with your library.

```
GET https://graph.microsoft.com/v1.0/me/mailboxSettings
```

**Comment by sreeise** [comment]
Hey @M-Patrone,

For sure I can help get you set up for that. Generating the code is a bit difficult to get setup so I am not sure you will want to go through that process given it will take a good bit of time. I have been trying to get a better setup for that for a while now but it all depends on the time I have and right now I am pretty busy so I apologize.

What you can do is basically add that request to the /me section. You can add it under this guy https://github.com/sreeise/graph-rs/blob/3cb16fba06cea812410a0fd5f74127ce89a529bf/src/me/request.rs#L742-L749

And just follow the same format as that request and then open up a PR and ill approve/merge it.

Is /me/mailboxSettings the only request your wanting to add?

**Reply by M-Patrone** [reply]
Thank you for your prompt response. Please don't apologize, everything is all right. I am truly grateful for the existence of the library.

At the moment, I am also interested in making the request `PATCH /me/mailboxSettings`. Additionally, there may be a need in the future for the request `PATCH/GET /users/{id|userPrincipalName}/mailboxSettings`.

If it is challenging to implement these requests, we do not have to prioritize them immediately. Perhaps if you have time later on, we can explore this further. However, it is not necessary.

Thank you for your help!

**Reply by sreeise** [reply]
So in that case we would want to generate a client to handle all /mailboxSettings paths so that any resource such as /me or /users can link to that client and use the path.

Not sure when you need this but I have a large change coming in around 3-4 weeks to the library for #392 and I can possibly see about adding /mailboxSettings paths to that branch.

**Reply by M-Patrone** [reply]
Sure, that sounds like a great plan! If I have some spare time, I'll try to add the feature myself and create a pull request. However, there's no need to rush as this is just a side project and I don't require the feature immediately. Thank you for your assistance, and please let me know if there's anything else I can do to help.

**Reply by sreeise** [reply]
Will do. Thanks for being understanding and patient as well 💯

**Reply by sreeise** [reply]
@M-Patrone I will be merging [this branch (async_default)](https://github.com/sreeise/graph-rs/tree/async_default) soon to move everything to async by default. This branch also has the mailboxSettings API in it. You can try out that branch in your dependencies if you want but there may be a few changes you have to make. From the user perspective the changes are minimal however,

I added an[ example](https://github.com/sreeise/graph-rs/blob/async_default/examples/mail_folders_and_messages/mailbox_settings.rs) to that branch so its easy to see what to do: 
https://github.com/sreeise/graph-rs/blob/781e266617c071e7a4d8fb99d32b963f66eaceae/examples/mail_folders_and_messages/mailbox_settings.rs#L1-L42

**Reply by sreeise** [reply]
1.0.0 has been merged. Let me know if you run into any issues 🚀

**Reply by M-Patrone** [reply]
Thank you really much @sreeise. I also wanted to apologize for not being able to assist you earlier. I had a lot of schoolwork to complete. I will try it as soon as possible!

**Reply by sreeise** [reply]
> Thank you really much @sreeise. I also wanted to apologize for not being able to assist you earlier. I had a lot of schoolwork to complete. I will try it as soon as possible!

No problem at all. Thanks for using the crate/project!
